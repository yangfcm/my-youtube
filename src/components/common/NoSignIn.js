import React from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import GoogleAuth from "../modules/GoogleAuth";

const NoSignIn = ({ googleSignin }) => {
  return (
    <div className="d-flex flex-column align-items-center">
      <div className="mb-4"></div>
      <h5 className="text-danger my-3"> You have not signed in </h5>
      <div style={{ height: "55px" }} className="d-flex align-items-center">
        <button className="btn btn-primary" onClick={googleSignin}>
          <FontAwesomeIcon icon="sign-in-alt" />
          {"   "}
          Sign In
        </button>
      </div>
    </div>
  );
};

export default GoogleAuth(NoSignIn);
