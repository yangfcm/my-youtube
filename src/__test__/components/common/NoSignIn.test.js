import React from "react";
import { Provider } from "react-redux";
import { mount } from "enzyme";
import configMockStore from "redux-mock-store";
import NoSignIn from "components/common/NoSignIn";

describe("Test NoSignIn component", () => {
  let wrapper, mockStore;
  const googleSigninSpy = jest.fn();
  beforeEach(() => {
    mockStore = configMockStore([]);
    wrapper = mount(<NoSignIn googleSignin={googleSigninSpy} />, {
      wrappingComponent: Provider,
      wrappingComponentProps: { store: mockStore({}) },
    });
  });
  it("should render NoSignIn component", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("should have no-signin message", () => {
    expect(wrapper.find("h5").text()).toContain("You have not signed in");
  });
  it("should have Sign in button and correct button text", () => {
    expect(wrapper.find("button").exists()).toBeTruthy();
    expect(wrapper.find("button").text()).toContain("Sign In");
  });
  it("should call googleSignin function when Sign in button is clicked", () => {
    wrapper.find("button").simulate("click");
    expect(googleSigninSpy).toHaveBeenCalled();
  });
});
