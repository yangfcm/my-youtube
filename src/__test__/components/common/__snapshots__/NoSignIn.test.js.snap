// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test NoSignIn component should render NoSignIn component 1`] = `
<Connect(GoogleAuthWrapper)
  googleSignin={[MockFunction]}
>
  <GoogleAuthWrapper
    googleSignin={[MockFunction]}
    signIn={[Function]}
    signOut={[Function]}
  >
    <NoSignIn
      googleSignin={[MockFunction]}
      googleSignout={[Function]}
      googleUser={null}
      signIn={[Function]}
      signOut={[Function]}
    >
      <div
        className="d-flex flex-column align-items-center"
      >
        <div
          className="mb-4"
        />
        <h5
          className="text-danger my-3"
        >
           You have not signed in 
        </h5>
        <div
          className="d-flex align-items-center"
          style={
            Object {
              "height": "55px",
            }
          }
        >
          <button
            className="btn btn-primary"
            onClick={[MockFunction]}
          >
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon="sign-in-alt"
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              swapOpacity={false}
              symbol={false}
              title=""
              transform={null}
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-sign-in-alt fa-w-16 "
                data-icon="sign-in-alt"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M416 448h-84c-6.6 0-12-5.4-12-12v-40c0-6.6 5.4-12 12-12h84c17.7 0 32-14.3 32-32V160c0-17.7-14.3-32-32-32h-84c-6.6 0-12-5.4-12-12V76c0-6.6 5.4-12 12-12h84c53 0 96 43 96 96v192c0 53-43 96-96 96zm-47-201L201 79c-15-15-41-4.5-41 17v96H24c-13.3 0-24 10.7-24 24v96c0 13.3 10.7 24 24 24h136v96c0 21.5 26 32 41 17l168-168c9.3-9.4 9.3-24.6 0-34z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </FontAwesomeIcon>
               
            Sign In
          </button>
        </div>
      </div>
    </NoSignIn>
  </GoogleAuthWrapper>
</Connect(GoogleAuthWrapper)>
`;
